ACE/ace/DLL_Manager.cpp

```diff
@@ -177,11 +178,19 @@ ACE_DLL_Handle::open (const ACE_TCHAR *dll_name,
               if (ACE_TString::npos != name->strstr (ACE_TEXT (".a")))
                 {
                   ACE_TCHAR aix_pathname[MAXPATHLEN + 1];
-                  ACE_OS::strncpy (aix_pathname,
-                                   name->c_str (),
-                                   name->length ());
-                  aix_pathname[name->length ()] = '\0';
-                  ACE_OS::strcat (aix_pathname, ACE_TEXT ("(shr.o)"));
+                  if (ACE_OS::snprintf (aix_pathname, sizeof (aix_pathname),
+                                        ACE_TEXT ("%s(shr.o)"),
+                                        name->c_str ()) < sizeof (aix_pathname))
+                    {
+
+                      if (ACE::debug ())
+                        ACELIB_ERROR ((LM_ERROR,
+                              ACE_TEXT ("ACE (%P|%t) DLL_Handle::open: ")
+                              ACE_TEXT ("('%s(shr.o)') is too long\n"),
+                              name->c_str()));
+
```

Comment:
Shouldn't we put an error text also to `errors` so that in case the caller uses `errors` he can log this error case at a higher level?